src/_pytest/doctest.py

```diff
@@ -465,15 +471,19 @@ def _find_lineno(self, obj, source_lines):
                 return doctest.DocTestFinder._find_lineno(self, obj, source_lines)
 
             def _find(
-                self, tests, obj, name, module, source_lines, globs, seen
+                _self, tests, obj, name, module, source_lines, globs, seen
             ) -> None:
                 if _is_mocked(obj):
                     return
+
+                if self.config.getini("doctest_standalone_namespace"):
+                    globs = {}
```

Comment:
Based on https://github.com/pytest-dev/pytest/blob/5.3.1/src/_pytest/doctest.py#L109-L113 that seems to use `DoctestModule` for `*.py` and `DoctestTextfile` for `*.txt`/`*.rst`, as this code change is only in `DoctestModule`, does this mean that this flag would apply only to the doctest examples inside `*.py` files and not the files that are fully doctest e.g. `*.rst`? Is that intended?

Reply:
I don't think non-py files can have variables that are defined outside of the doctest. Or is there something I'm missing?

Reply:
I'm not sure how this currently interacts with that, nor how it should.

Reply:
It does interact, but in this case I don't think the new option should interact at all with the `doctest_namespace` fixture: the intent there is to inject globals into the tests (such as the `numpy` import), so resetting that would defeat the purpose.

Reply:
Maybe it makes sense to allow `globs` to be modified in a fixture in conftest.py (side note: "globs" is an unfortunate name from doctest; it is short for "globals" and has nothing to do with shell globs). Although I do like the simplicity of an ini setting to get the `globs = {}` behavior.

Reply:
Actually, taking a second look at the code, `doctest_namespace` won't be affected by this at all, because the `doctest_namespace` dict is injected during the `setup` phase of the test, while here `globs` is cleared during collection.

@asmeurer could you please add a test which ensures setting `doctest_standalone_namespace` to True still allow doctests to access the dict returned by `doctest_namespace`? We also should explicitly mention this in the description of the `doctest_standalone_namespace`.